fab/trainable_distributions/defensive_mixture.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional, Tuple
```

Comment:
I added this earlier on my branch as I think it may improve things, but currently it is not used anywhere and I need to still do more thorough testing of it.